### PR TITLE
[CAT-564] Up and running tutorial

### PR DESCRIPTION
**Pull request by jlyon1**

### CODE REVIEW

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+3. Request a Match
+4. Connect to a Server
+5. Play a game
+
+## Prerequisites
+
+What you will need before following the steps in this guide.
+
+* Access to Catena Tools
+* A Game with a client server model or the [Catena Lyra Demo](https://github.com/catenaTools/catena-lyra-demo)
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: A game with a client/dedicated game server model.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+* The Catena Tools [AWS Infrastructure Template](https://github.com/CatenaTools/infrastructure/tree/main/aws/catena-core)
+* [Terraform](https://www.terraform.io) V1.71 or higher
+* An AWS account
+* Git bash (optional)
+* A domain name that you control and can configure in route53
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: ...and can configure in [AWS Route 53](https://aws.amazon.com/route53/)

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+
+## Setup the Infrastructure
+
+Prior to running the Terraform code, set up a hosted zone for the domain that you will use. In the example, this will be `catenatools.com.`
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Prior to provisioning any infrastructure, you will need to set up a hosted zone in Route 53 for the domain that you will use.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+
+See [Creating a public hosted zone](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/CreatingHostedZone.html). The Terraform configuration will create a few subdomain routes on this url.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: The Terraform configuration in the AWS Infrastructure Template will create a few subdomain routes on this URL.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+
+Next, clone the `infrastructure` repo and open it in your favorite editor. We will be working out of the `aws/catena-core/` directory.
+
+**Git Bash**
+```bash
+git clone git@github.com:CatenaTools/infrastructure.git
+cd infrastructure/aws
+```
+
+This template sets up a working [dokku](https://dokku.com) instance on the domain you select, along with a few additional settings to simplify deploying and running the platform, including an IAM role for `aws ssm` logins and an Elastic IP.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md:49]: Maybe add another line explaining why we use Dokku?

This Catena deployment utilizes Dokku to make deploying updates as simple as `git push`. You don't need to worry too much about all of the infrastructure if you don't want to. This tutorial will outline the pieces that are important for you to understand in order to get up and running.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+
+### Configure Terraform
+
+The Terraform configuration defaults to using an s3 bucket to store it's state, this must be updated to an s3 bucket on your account. You may create this s3 bucket in the AWS Console.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Capitalize: S3

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+```
+  backend "s3" {
+      bucket  = "catena-terraform-state"
+      key     = "updated-infra/terraform.tfstate"
+      region  = "us-east-1"
+      profile = "catena_admin"
+  }
+```
+
+Be sure to replace `bucket` with your bucket's name, `region` with your region, and `profile` with your local [aws profile](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configure/index.html).
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Capitalize: AWS

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+
+> If you do not need to back up or share your terraform state, you can remove the "backend" block and terraform will store it's state locally.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Capitalize: Terraform

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 9]
```diff
@@ -0,0 +1,528 @@
...
+git clone git@github.com:CatenaTools/catena-tools-core.git
+cd catena-tools-core/
+```
+
+Then add a new remote and do a git push, this will deploy the `HEAD` of the repo. The command to do so is the value of the `add_doku_remote_command` output.
+
+**Git Bash**
+```bash
+git remote add dokku dokku@dev.catenatools.com:platform
+git push dokku HEAD:main
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: If for some reason a developer uses a key _not_ called `id_rsa`, they'll need to use the relevant `deploy_command` from the Terraform output.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 10]
```diff
@@ -0,0 +1,528 @@
...
+We will be running our game on an AWS EC2 Instance running Windows Server, these instructions also work with a Linux server using minor modifications.
+
+First create a key pair in the AWS console for remote access to the machine:
+
+![access_key_quickstart](access_key_quickstart.png)
+
+Next, configure terraform to provision an ec2 instance where we will run the server, this should be in `ec2/main.tf`
+
+```
+resource aws_instance "server_instance" {
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Carrying comment over from last PR. We should likely just provide this as another directory in the `infrastructure` repo rather than requiring copy/paste. I can work on that after I'm done with GameLift stuff.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 11]
```diff
@@ -0,0 +1,528 @@
...
+```
+
+Next plan and apply your infrastructure
+
+```
+terraform plan -var-file="vars.tfvar" 
+terraform apply -var-file="vars.tfvar"
+```
+
+Once terraform finishes, you can go to the AWS EC2 Console, click your instance, and click "Connect to instance." Under the RDP tab, download the
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Capitalize: Terraform

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 12]
```diff
@@ -0,0 +1,528 @@
...
+}
+```
+
+The key is to set the following fields:
+
+`app.backend.catena_backend.url` field, which must point to your deployed instance url.
+`app.resolver.configs.ec2_resolver.Min` - The min port to use for clients to connect.
+`app.resolver.configs.ec2_resolver.Max` - The max port to use for clients to connect.
+
+_For this demo, min and max port should be the same, and be the port your server runs on. If you are following this demo, it should be `7777`
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Missing closing `_`

**Review by hunterfortuin — COMMENTED** [review]
Looking good! I have a few minor nitpicks since this is public facing.

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 10 above)
Comment (jlyon1) [reply Writerside/topics/Quickstart-Up-And-Running.md]: Sounds good, i've put it in a ticket here: https://catenatools.atlassian.net/browse/CAT-567

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 13]
```diff
@@ -0,0 +1,586 @@
...
+4. Connect to a Server
+5. Play a game
+
+## Prerequisites
+
+What you will need before following the steps in this guide.
+
+* Access to Catena Tools
+* A Game with a client/dedicated game server model or the [Catena Lyra Demo](https://github.com/catenaTools/catena-lyra-demo)
+* The Catena Tools [AWS Infrastructure Template](https://github.com/CatenaTools/infrastructure/tree/main/AWS/catena-core)
```
Comment (academician) [inline Writerside/topics/Quickstart-Up-And-Running.md]: This link looks broken - should be https://github.com/CatenaTools/infrastructure/tree/main/aws/catena-core

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 14]
```diff
@@ -0,0 +1,585 @@
...
+
+This template sets up a working [dokku](https://dokku.com) instance on the domain you select, along with a few additional settings to simplify deploying and running the platform, including an IAM role for `aws ssm` logins and an Elastic IP.
+
+> We set things up this way to make deploying your Catena instance as simple as running `git push`. You don't need to worry too much about all of the infrastructure if you don't want to. This tutorial will outline the pieces that are important for you to understand in order to get up and running.
+
+### Configure Terraform
+
+The Terraform configuration defaults to using an S3 bucket to store it's state, this must be updated to an S3 bucket on your account. You may create this S3 bucket in the AWS Console.
+
+Open `terraform_config.tf` and modify the following portion to match your S3 bucket, or remove it to store state locally.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: "remove it" is ambiguous - the section or the file?

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 15]
```diff
@@ -0,0 +1,585 @@
...
+cd infrastructure/aws
+```
+
+This template sets up a working [dokku](https://dokku.com) instance on the domain you select, along with a few additional settings to simplify deploying and running the platform, including an IAM role for `aws ssm` logins and an Elastic IP.
+
+> We set things up this way to make deploying your Catena instance as simple as running `git push`. You don't need to worry too much about all of the infrastructure if you don't want to. This tutorial will outline the pieces that are important for you to understand in order to get up and running.
+
+### Configure Terraform
+
+The Terraform configuration defaults to using an S3 bucket to store it's state, this must be updated to an S3 bucket on your account. You may create this S3 bucket in the AWS Console.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Recommending a slight rephrasing so this reads more clearly with no advanced knowledge:

> The Terraform configuration defaults to using an S3 bucket to store it's state. You may create an S3 bucket in your account with the AWS Console and update the configuration or remove the S3 configuration to store state locally.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 16]
```diff
@@ -0,0 +1,585 @@
...
+      region  = "us-east-1"
+      profile = "catena_admin"
+  }
+```
+
+Be sure to replace `bucket` with your bucket's name, `region` with your region, and `profile` with your local [AWS Profile](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configure/index.html).
+
+> If you do not need to back up or share your terraform state, you can remove the "backend" block and terraform will store it's state locally.
+
+Next, we need to set some specific-to-you configurations.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: > set some specific-to-you configurations

Colloquialisms can make tutorials more informal (good IMO) but less accessible. We might want to rephrase this to "set some configurations specific to your game"

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 17]
```diff
@@ -0,0 +1,585 @@
...
+ec2_instance_id = "i-0f47e887113bf882b"
+ec2_instance_ssh_command = "aws ec2-instance-connect ssh --os-user ubuntu --instance-id i-0f47e887113bf882b --profile catena"
+ec2_ip = "100.28.26.117"
+powershell_deploy_command = "$env:GIT_SSH_COMMAND='ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes'; git push dokku main"
+unix_deploy_command = "GIT_SSH_COMMAND='ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes' git push dokku main"
+windows_deploy_command = "set \"GIT_SSH_COMMAND=ssh -i ~/.ssh/id_rsa -o IdentitiesOnly=yes\" && git push dokku main"
+catena_url = "https://platform.dev.catenatools.com"
+```
+
+These outputs will be used to deploy the platform to your newly configured instance.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md:109]: It might be worth stressing that these values should be saved somewhere for the rest of the setup or how to retrieve them. It's easy to lose track of things like this while following a guide for the first time. (Maybe we want to do something like write these out with `local_file` in the infrastructure?)

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 18]
```diff
@@ -0,0 +1,585 @@
...
+curl --location '$CATENA_URL/api/v1/accounts' \
+--header "session-id: $SESSION_ID" \
+--header 'Content-Type: application/json' \
+--data '{}'
+
+# {"account":{"id":"account-f378e3c2-8760-4059-ba91-69f5343dfb48","displayName":"test53","authRole":"user","metadata":{},"providers":["PROVIDER_UNSAFE"]}}
+```
+
+[//]: # (TODO: Add the link here.)
+The full postman collection can be found here. Set the http-host environment variable to your domain prior to use.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: This line shows up without the TODO and is confusing - maybe we comment it out until the collection is available?

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 19]
```diff
@@ -0,0 +1,585 @@
...
+
+## Running your game in a multiplayer configuration
+
+Next we will use the server [Sidecar](Sidecar.md) along with our backend to support a multiplayer workflow.
+
+> The sidecar enables you to configure your game to work with one API interface, and deploy it on multiple fleet managers. It can also make requests on behalf of the game server, enabling integrations which require zero server-side integration.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: >  enabling integrations which require zero server-side integration

This is somewhat confusing phrasing.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 20]
```diff
@@ -0,0 +1,585 @@
...
+    <p>These instructions use the <a href="https://github.com/catenatools/catena-lyra-demo">catena-lyra-demo</a> server.</p>
+    <step>Download the server onto the machine. The download can be found <a href="https://catena-public-content.S3.amazonaws.com/WindowsServer.zip">here.</a></step>
+    <step>Extract the zip file.</step>
+    <step>Run the game server from command prompt: <pre>LyraServer.exe -networkversionoverride=1234 -BackendUrl=localhost:8085</pre></step>
+    <step>Now the game server will communicate with the backend through the sidecar.</step>
+</procedure>
+
+## Run the Server Sidecar
+
+The server sidecar enables Catena to be aware of your game server without requiring any changes to the code of the game server itself. It is additionally able to make requests on behalf of your game server
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: > It is additionally able to make

"additionally" is modifying "it" (server sidecar) and reads a little better at the beginning of the sentence IMO:

> Additionally, it is able to make

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 21]
```diff
@@ -0,0 +1,585 @@
...
+if you so choose. The sidecar will augment connection details and can translate requests between different backends so your server only needs to implement one SDK interface. See [Sidecar](Sidecar.md) for more details.
+
+First download the Sidecar to the machine, the latest release can be found [here](https://github.com/CatenaTools/sidecar/releases/). Alternatively, clone and build the sidecar.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: > the machine

Their machine? The game server machine?

Comment (doctaweeks) [reply Writerside/topics/Quickstart-Up-And-Running.md]: It might be good to be more specific about downloading the precompiled windows binary since there are multiple files for the release.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 22]
```diff
@@ -0,0 +1,585 @@
...
+
+In the same directory as the sidecar, add a `config.json.` There is an example in the  sidecar repo here: https://github.com/CatenaTools/sidecar/blob/main/config.json
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Being more specific about "the sidecar" here may be good - this should be adjacent to the compiled binary. (User may download/clone and build themselves.)

Comment (doctaweeks) [reply Writerside/topics/Quickstart-Up-And-Running.md]: Missing hyperlink

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 23]
```diff
@@ -0,0 +1,585 @@
...
+        "AutoPopulateIDs": true
+      }
+    }
+  }
+}
+```
+
+The key is to set the following fields:
+
+`app.backend.catena_backend.url` field, which must point to your deployed instance url.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: > field, which

leftover from a previous version? Reads a little odd and is different from the subsequent lines.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 24]
```diff
@@ -0,0 +1,585 @@
...
+`app.resolver.configs.ec2_resolver.Min` - The min port to use for clients to connect.
+`app.resolver.configs.ec2_resolver.Max` - The max port to use for clients to connect.
+
+_For this demo, min and max port should be the same, and be the port your server runs on. If you are following this demo, it should be `7777`_
+
+> Also set is the `app.event_source.use` field, which is `catena_api_event_source.` If your server implements the [Catena Server Manager](https://github.com/CatenaTools/catena-tools-core/blob/main/Protos/api/v1/catena_server_manager.proto) interface
+> it can make requests to this backend.
+
+After which you can launch the sidecar using `cmd.exe`
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md:345]: > using `cmd.exe`

Some users may need more clarity/specific instructions.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 25]
```diff
@@ -0,0 +1,585 @@
...
+> If you restart the sidecar with this config, you will see it requesting matches repeatedly.
+> 
+> See [Match Broker](Match-Broker.md) to read about how Catena manages servers.
+
+
+## Testing it out
+
+Next we will download the sample game and test it out. In this case we will use a prebuilt version of the catena lyra demo. It can be found [here](https://catena-public-content.s3.amazonaws.com/WindowsClient.zip).
+
+So long as the server is still running from the last example`, we simply need to launch the game with a few arguments, and we're good to go.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Stray tick `

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 26]
```diff
@@ -0,0 +1,585 @@
...
+
+# session-ca8fb665-69df-462f-941e-3341275fd91c
+```
+
+Launch the game
+
+```
+LyraGame.exe -BackendURL $CATENA_URL -session-id $SESSION_ID -networkversionoverride=1234
+```
+From here, you can navigate through Play Lyra -> Matchmaking -> Elimination. You should see your game enter matchmkaing.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md]: typo: "matchmkaing"

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 17 above)
Comment (jlyon1) [reply Writerside/topics/Quickstart-Up-And-Running.md:109]: This is a great idea! Maybe even create a source-able file?

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 19 above)
Comment (jlyon1) [reply Writerside/topics/Quickstart-Up-And-Running.md]:
```suggestion
> The sidecar enables you to configure your game to work with one API interface, and deploy it on multiple fleet managers. It can also make requests on behalf of the game server reducing the burden to set up a dedicated game server.
```

**Comment by jlyon1** [comment]
@doctaweeks integrated your changes. Going to do the local file suggestion in a separate PR: https://catenatools.atlassian.net/browse/CAT-571

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 10 above)
Comment (hunterfortuin) [reply Writerside/topics/Quickstart-Up-And-Running.md]: Open PR here: https://github.com/CatenaTools/infrastructure/pull/9

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 27]
```diff
@@ -0,0 +1,591 @@
...
+```bash
+git clone git@github.com:CatenaTools/catena-tools-core.git
+cd catena-tools-core/
+```
+
+Then add a new remote and do a git push, this will deploy the `HEAD` of the repo. The command to do so is the value of the `add_doku_remote_command` output. The push command is platform specific, use the `{powershell,unix,windows}_deploy_command` output above.
+
+**Git Bash**
+```bash
+git remote add dokku dokku@dev.catenatools.com:platform
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Until we set up the source file, we'll want to call out when commands will need to be swapped out. I.e.:

```
git remote add dokku dokku@dev.catenatools.com:platform ## Replace this with the "add_dokku_remote_command" output by Terraform
git push dokku HEAD:main ## Replace this with your platform's deploy command
```

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 28]
```diff
@@ -0,0 +1,591 @@
...
+
+curl --location '$CATENA_URL/api/v1/accounts' \
+--header "session-id: $SESSION_ID" \
+--header 'Content-Type: application/json' \
+--data '{}'
+
+# {"account":{"id":"account-f378e3c2-8760-4059-ba91-69f5343dfb48","displayName":"test53","authRole":"user","metadata":{},"providers":["PROVIDER_UNSAFE"]}}
+```
+
+The full postman collection can be found [here](https://github.com/CatenaTools/catena-tools-core/tree/main/scripts). Set the http-host environment variable to your domain prior to use.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: postman -> Postman

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 29]
```diff
@@ -0,0 +1,591 @@
...
+> every 5 seconds. So long as your server is listening on port 7777 clients will be able to connect to your server.
+> 
+> If you restart the sidecar with this config, you will see it requesting matches repeatedly.
+> 
+> See [Match Broker](Match-Broker.md) to read about how Catena manages servers.
+
+
+## Testing it out
+
+Next we will download the sample game and test it out. In this case we will use a prebuilt version of the catena lyra demo. It can be found [here](https://catena-public-content.s3.amazonaws.com/WindowsClient.zip).
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: catena lyra demo -> Catena Lyra Demo

(The whole tutorial should get a pass on capitalization. Not the end of the world, but is good for correctness)

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 30]
```diff
@@ -0,0 +1,591 @@
...
+"provider": "PROVIDER_UNSAFE",
+"payload": "test01"
+}' | grep "session-*" | cut -d' ' -f2)
+
+curl --location '$CATENA_URL/api/v1/accounts' \
+--header "session-id: $SESSION_ID_1" \
+--header 'Content-Type: application/json' \
+--data '{}'
+
+echo $SESSION_ID
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: We're assigning `SESSION_ID_1` but querying `SESSION_ID` here.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 31]
```diff
@@ -0,0 +1,591 @@
...
+
+```
+SESSION_ID_1=$(curl -D - -sS --location '$CATENA_URL/api/v1/authentication/login' \
+--header 'Content-Type: application/json' \
+--data '{
+"provider": "PROVIDER_UNSAFE",
+"payload": "test01"
+}' | grep "session-*" | cut -d' ' -f2)
+
+curl --location '$CATENA_URL/api/v1/accounts' \
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md:390]: May be worth a note on why this second call is necessary.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 32]
```diff
@@ -0,0 +1,591 @@
...
+
+echo $SESSION_ID
+
+# session-ca8fb665-69df-462f-941e-3341275fd91c
+```
+
+Launch the game
+
+```
+LyraGame.exe -BackendURL $CATENA_URL -session-id $SESSION_ID -networkversionoverride=1234
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Do we have a configuration we can run Lyra in that will automatically obtain a session rather than requiring manually obtaining it ahead of time?

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 33]
```diff
@@ -0,0 +1,591 @@
...
+Remote Desktop File. Then click "Get Password" and paste in the private key from the keypair you created previously.
+
+At this point you can connect to the Windows server using RDP by opening the file and logging into the machine with the "Administrator" account and password
+from the AWS console.
+
+At this point you should download and prepare to run your game on the server. Instructions for the Lyra Demo are below.
+
+<procedure title="Lyra Game Server Installation" id="lyra_setup" collapsible="true">
+    <p>These instructions use the <a href="https://github.com/catenatools/catena-lyra-demo">catena-lyra-demo</a> server.</p>
+    <step>Download the server onto the machine. The download can be found <a href="https://catena-public-content.S3.amazonaws.com/WindowsServer.zip">here.</a></step>
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md:255]: Last big comment that just occurred to me. We may want to provide some instructions for copying local files to their remote desktop/remote machine if they're not going to be using Lyra.

Linux: `scp`
Windows: https://serverfault.com/questions/150369/how-to-copy-files-to-amazon-ec2-windows-instance-from-my-local-machine

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 32 above)
Comment (jlyon1) [reply Writerside/topics/Quickstart-Up-And-Running.md]: it actually _should_ do this by default

**Comment by jlyon1** [comment]
@hunterfortuin @doctaweeks this should encompass all feedback now. let me know what you both think

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 34]
```diff
@@ -0,0 +1,562 @@
...
+Prior to provisioning any infrastructure, you will need to set up a hosted zone in Route 53 for the domain that you will use. In the example, this will be `catenatools.com.`
+
+See [Creating a public hosted zone](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/CreatingHostedZone.html). The Terraform configuration in the AWS Infrastructure Template will create a few subdomain routes on this URL.
+
+Next, clone the `infrastructure` repo and open it in your favorite editor. We will be working out of the `aws/catena-core/` directory.
+
+**Git Bash**
+```bash
+git clone git@github.com:CatenaTools/infrastructure.git
+cd infrastructure/aws
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: `cd infrastructure/aws/catena-core/`

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 35]
```diff
@@ -211,63 +211,17 @@ First create a key pair in the AWS console for remote access to the machine:
 
 ![access_key_quickstart](access_key_quickstart.png)
 
-Next, configure terraform to provision an ec2 instance where we will run the server, this should be in `ec2/main.tf`
+Next, configure terraform to provision an ec2 instance where we will run the server. we have provided a template for this in the `aws/ec2-gameserver` directory of the infrastructure repo.
```
Comment (doctaweeks) [inline Writerside/topics/Quickstart-Up-And-Running.md:225]: typo: "we" -> "We"

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 36]
```diff
@@ -0,0 +1,562 @@
...
+1. The server will be "always open" and accepting players
+2. There will be only one type of match
+
+### Setup our Game Server
+
+We will be running our game on an AWS EC2 Instance running Windows Server, these instructions also work with a Linux server using minor modifications.
+
+First create a key pair in the AWS console for remote access to the machine:
+
+![access_key_quickstart](access_key_quickstart.png)
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: We create this AWS keypair in the [Terraform files](https://github.com/CatenaTools/infrastructure/blob/main/aws/ec2-gameserver/ec2/main.tf#L31).

If an end user needs a local keypair to seed this, they should generate one on their machine (works on both Windows and Linux):
```
ssh-keygen -t rsa -b 2048 -m PEM -f my_private_key_file
```

They can then find the keys to pass in to the tfvar's...
```
my_private_key_file
my_private_key_file.pub
```

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 37]
```diff
@@ -0,0 +1,562 @@
...
+```
+
+Next plan and apply your infrastructure
+
+```
+terraform plan -var-file="vars.tfvar" 
+terraform apply -var-file="vars.tfvar"
+```
+
+Once Terraform finishes, you can go to the AWS EC2 Console, click your instance, and click "Connect to instance." Under the RDP tab, download the
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md:245]: They can also run `terraform output -raw ec2_windows_password` to get the password instead of copy/pasting the private key.

File: Writerside/topics/Quickstart-Up-And-Running.md [hunk 38]
```diff
@@ -0,0 +1,562 @@
...
+Remote Desktop File. Then click "Get Password" and paste in the private key from the keypair you created previously.
+
+At this point you can connect to the Windows server using RDP by opening the file and logging into the machine with the "Administrator" account and password
+from the AWS console.
```
Comment (hunterfortuin) [inline Writerside/topics/Quickstart-Up-And-Running.md]: Erroneous newline

**Review by hunterfortuin — COMMENTED** [review]
At the risk of going overboard (I think we could go back and forth refining this for a while), I think this is my last round of feedback and then I'm of the opinion that we should ship it and capture any future updates as new work in a subsequent PR.

File: Writerside/topics/Quickstart-Up-And-Running.md (hunk 36 above)
Comment (jlyon1) [reply Writerside/topics/Quickstart-Up-And-Running.md]: using the password command means we don't need this, so I will remove it instead.

Comment (hunterfortuin) [reply Writerside/topics/Quickstart-Up-And-Running.md]: The keys are still necessary to provision everything via Terraform. It _could_ just be inherent in the instructions we give regarding the tfvars though.
